src/device-registry/utils/create-event.js

```diff
@@ -17,12 +17,119 @@ const axios = require("axios");
 const { kafkaConsumer } = require("../config/kafkajs");
 const mongoose = require("mongoose");
 const ObjectId = mongoose.Types.ObjectId;
-
-const { generateDateFormat, generateDateFormatWithoutHrs } = require("./date");
+const { BigQuery } = require("@google-cloud/bigquery");
+const bigquery = new BigQuery();
+const {
+  generateDateFormat,
+  threeMonthsFromNow,
+  generateDateFormatWithoutHrs,
+  addMonthsToProvideDateTime,
+  monthsInfront,
+  isTimeEmpty,
+  getDifferenceInMonths,
+  addDays,
+  addMinutes,
+} = require("./date");
 
 const httpStatus = require("http-status");
 
 const createEvent = {
+  getMeasurementsFromBigQuery: async (req) => {
+    try {
+      const { query } = req;
+      const {
+        frequency,
+        device,
+        name,
+        startTime,
+        endTime,
+        tenant,
+        limit,
+        skip,
+        site,
+      } = query;
+
+      /**
+       *
+       * 1. We now just have to test the datetime
+       * things and we shall be good to go
+       *
+       * 2. We also need to embrace the usage of ENVs for
+       * some of the variables below
+       */
+
+      const currentDate = generateDateFormatWithoutHrs(new Date());
+
+      const twoMonthsBack = generateDateFormatWithoutHrs(
+        addMonthsToProvideDateTime(currentDate, -2)
+      );
+
+      const start = generateDateFormatWithoutHrs(
+        startTime ? endTime : twoMonthsBack
+      );
+      const end = generateDateFormatWithoutHrs(endTime ? endTime : currentDate);
+
+      let table = "airqo-250220.averaged_data_stage.hourly_device_measurements";
+      let pm2_5 = "";
+      let pm10 = "";
+
+      if (frequency === "raw") {
+        table = "airqo-250220.raw_data.device_measurements";
+        pm2_5 = "";
+        pm10 = "";
+      }
+
+      const queryStatement = `SELECT site_id, name, device, \`airqo-250220.metadata_stage.sites\`.latitude AS latitude,
+        \`airqo-250220.metadata_stage.sites\`.longitude AS longitude, timestamp, pm2_5, pm10, pm2_5_raw_value, pm2_5_calibrated_value, pm10_raw_value, pm10_calibrated_value,
+        \`airqo-250220.metadata_stage.sites\`.tenant AS tenant 
+        FROM \`${table}\` 
+        JOIN \`airqo-250220.metadata_stage.sites\` 
+        ON \`airqo-250220.metadata_stage.sites\`.id = \`${table}\`.site_id 
+        WHERE timestamp  
+       >= "${start ? start : twoMonthsBack}" AND timestamp <= "${
+        end ? end : currentDate
+      }" 
+      ${site ? `AND site_id="${site}"` : ""}
+      ${device ? `AND device="${device}"` : ""}
+      ${
+        tenant
+          ? `AND \`airqo-250220.metadata_stage.sites\`.tenant="${tenant}"`
+          : ""
+      }
+        LIMIT ${limit ? limit : constants.DEFAULT_EVENTS_LIMIT}`;
```

Comment:
Hi @Baalmart this query is most likely to fail because;

1. The the datasets and tables are not the right production datasets and tables. Consider using environment variables.
2. Not all tables have columns for calibrated values.

Reply:
Hi @NoahNsimbe , thanks for the suggestion of using ENVs, shall review and raise new commits as soon as possible.
Thanks!

Reply:
Hi @Baalmart staging and production datasets have the same structure. Using ENVs can help to solve the problem of environments.
Great, if all tables have calibrated data. The query shall not fail.

Just a side note, is that we have approximated coordinates for sites. Could we switch to using them. Reference : #895